Jenkinsfile

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env groovy
+
+pipeline {
+    agent { docker defaultWorker.getConfig("openstack-test") }
+    options {
+        ansiColor('xterm')
+        timestamps()
+        timeout(time: 2, unit: "HOURS")
+    }
+    stages {
+        stage("systest") {
+            steps {
+                sh '''
+                    # - initialize job control vars
+                    # - JOB_NAME is an environment variable that is
+                    #   automatically provided by Jenkins
+                    #   (eg. "openstack/agent/mitaka/12_1_1-overcloud")
+                    export PROGRAM=$(echo $JOB_NAME | cut -d "/" -f 1)
+                    export PROJECT=$(echo $JOB_NAME | cut -d "/" -f 2)
+                    export BRANCH=$(echo $JOB_NAME | cut -d "/" -f 3)
+                    export SYSTEST_JOB=$(echo $JOB_NAME | cut -d "/" -f 4)
+
+                    # - print build properties
+                    printenv | sort | grep -v OS_PASSWORD
+
+                    # - setup ssh agent
+                    eval $(ssh-agent -s)
+                    ssh-add
+
+                    # - run tests
+                    if [ "$SYSTEST_JOB" = "unit-tests" ]; then
+                        make -C systest unit-tests
+                    else
+                        export TESTENV_PATH="/usr/local/share/testenv"
+                        export BIGIP=$(echo $SYSTEST_JOB | cut -d "-" -f 1)
+                        export CLOUD=$(echo $SYSTEST_JOB | cut -d "-" -f 2)
```

Comment:
Now that you've made me actually write down my reasons for doing what I did, I'm starting to feel like I should just give you the more scalable solution (ie. properties files), right off the bat.  Will have to get back to you on that.